core/coreapi/interface/dag.go

```diff
@@ -4,21 +4,38 @@ import (
 	"context"
 	"io"
 
-	options "github.com/ipfs/go-ipfs/core/coreapi/interface/options"
+	"github.com/ipfs/go-ipfs/core/coreapi/interface/options"
 
 	ipld "gx/ipfs/QmZtNq8dArGfnpCZfx2pUNY7UcjGhVp5qqwQ4hH6mpTMRQ/go-ipld-format"
 )
 
-// DagAPI specifies the interface to IPLD
-type DagAPI interface {
+// DagOps groups operations that can be batched together
+type DagOps interface {
 	// Put inserts data using specified format and input encoding.
 	// Unless used with WithCodec or WithHash, the defaults "dag-cbor" and
 	// "sha256" are used.
 	Put(ctx context.Context, src io.Reader, opts ...options.DagPutOption) (ResolvedPath, error)
+}
+
+// DagBatch is the batching version of DagAPI. All implementations of DagBatch
+// should be threadsafe
+type DagBatch interface {
+	DagOps
+
+	// Commit commits nodes to the datastore and announces them to the network
+	Commit(ctx context.Context) error
```

Comment:
Note: need this context for go-ipfs-api implementation (batch will hold a connection and stream files to `dag/put`)